File: src/main/java/ai/aimachineserver/application/AiCallService.java
package ai.aimachineserver.application;

import ai.aimachineserver.application.commands.CallAiCommand;
import ai.aimachineserver.config.AppConfig;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class AiCallService {
    private final RestTemplate restTemplate;
    private final AppConfig appConfig;

    @Autowired
    public AiCallService(RestTemplate restTemplate, AppConfig appConfig) {
        this.restTemplate = restTemplate;
        this.appConfig = appConfig;
    }

    public void callAi(CallAiCommand command) {
        String requestUrl = String.format("%s/%s?requestedGameType=%s&gameId=%s",
                appConfig.aiServiceUrl,
                command.gameName,
                command.requestedGameType,
                command.gameId
        );
        ResponseEntity<String> response = restTemplate.getForEntity(requestUrl, String.class);
        if (response.getStatusCode() != HttpStatus.CREATED) {
            System.out.println("Unexpected status code while calling AI");
        }
    }
}

File: src/main/kotlin/ai/aimachineserver/application/AiCallService.kt
package ai.aimachineserver.application

import ai.aimachineserver.config.AppConfig
import ai.aimachineserver.application.commands.CallAiCommand
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate

@Service
class AiCallService(
    private val restTemplate: RestTemplate,
    private val appConfig: AppConfig
) {
    fun callAi(command: CallAiCommand): ResponseEntity<String> {
        val requestUrl =
            "${appConfig.aiServiceUrl}/${command.gameName}?requestedGameType=${command.requestedGameType}&gameId=${command.gameId}"
        return restTemplate.getForEntity(requestUrl, String::class.java)
    }
}
